indirect.h

```diff
@@ -45,6 +45,34 @@ namespace xyz {
 }
 #endif  // XYZ_UNREACHABLE_DEFINED
 
+namespace detail {
+template <class T, class U>
+concept three_way_comparable_with = requires(T& t, U& u) {
```

Comment:
What is the benefit of defining our own versions of these over using `std:: three_way_comparable`: https://en.cppreference.com/w/cpp/utility/compare/three_way_comparable

Also see, the three_way_comparable_with here matches `std:: three_way_comparable` and not `std:: three_way_comparable_with`

Reply:
I can't get `std:: three_way_comparable` to play nice. Issues with conversion of types and recursion I think.

I'd like a working implementation merged before plenary: can we follow up and improve code when you're with us? It will be easier to discuss in person.

Reply:
Yes, of course, only minor comments and not worth delaying for.

Reply:
It would be nice to sort out but we can merge for now.